src/ModelContextProtocol/Protocol/Types/GetPromptRequestParams.cs

```diff
@@ -16,5 +16,5 @@ public class GetPromptRequestParams : RequestParams
     /// Arguments to use for templating the prompt.
     /// </summary>
     [System.Text.Json.Serialization.JsonPropertyName("arguments")]
-    public Dictionary<string, object>? Arguments { get; init; }
+    public IReadOnlyDictionary<string, object?>? Arguments { get; init; }
```

Comment:
I missed this in my earlier changes, but this should also be `JsonElement` so that the relevant extension pre-serializers user-provided objects with potentially user-provided JSOs.
```suggestion
    public IReadOnlyDictionary<string, JsonElement>? Arguments { get; init; }
```